sklearn/metrics/pairwise.py

```diff
@@ -127,11 +128,33 @@ def check_pairwise_arrays(
         - 'allow-nan': accepts only np.nan and pd.NA values in array. Values
           cannot be infinite.
 
-        .. versionadded:: 0.22
+        .. versionadded:: 0.20
            ``force_all_finite`` accepts the string ``'allow-nan'``.
 
         .. versionchanged:: 0.23
-           Accepts `pd.NA` and converts it into `np.nan`.
+           Accepts `pd.NA` and converts it into `np.nan`
+
+        .. deprecated:: 1.6
+           `force_all_finite` was renamed to `ensure_all_finite` and will be removed
+           in 1.8.
+
+    ensure_all_finite : bool or 'allow-nan', default=True
+        Whether to raise an error on np.inf, np.nan, pd.NA in array. The
+        possibilities are:
+
+        - True: Force all values of array to be finite.
+        - False: accepts np.inf, np.nan, pd.NA in array.
+        - 'allow-nan': accepts only np.nan and pd.NA values in array. Values
+          cannot be infinite.
+
+        .. versionadded:: 0.20
+           Accepts the string ``'allow-nan'``.
+
+        .. versionchanged:: 0.23
+           Accepts `pd.NA` and converts it into `np.nan`
+
+        .. versionadded:: 1.6
+           `force_all_finite` was renamed to `ensure_all_finite`.
```

Comment:
we only need a version added I guess?

Reply:
The thing, if we don't copy it here, is that in 2 versions when we remove force_all_finite, then this information will disappear. I find it weird to remove the history for a renaming of a parameter.

Reply:
I don't think this information is relevant anymore when we're renaming the parameter.

Reply:
I agree with @adrinjalali. Basically since we change the name of parameter, we cannot go back to version 0.23 because this parameter is not existing there.

We just need to make sure that the options added are mentioned in the docstring that is the case here.

Reply:
Ok I removed that